File: ensate-rh-backend/src/main/java/ma/ensate/rh/ensate_rh_backend/dto/CitoyenDTO.java
package ma.ensate.rh.ensate_rh_backend.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ma.ensate.rh.ensate_rh_backend.model.Citoyen;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CitoyenDTO {
    
    @NotBlank(message = "CIN is required")
    private String cin;
    
    private String nomFr;
    
    private String nomAr;
    
    private String prenomFr;
    
    private String prenomAr;
    
    private String adresse;
    
    private String email;
    
    private String telephone;
    
    private LocalDate dateNaissance;
    
    private String lieuNaissance;
    
    private Citoyen.Sexe sexe;
    
    private String photoUrl;
}

File: FrontEnd/app/src/main/java/com/ensa/gestionpersonnel/data/remote/dto/CitoyenDto.kt
package com.ensa.gestionpersonnel.data.remote.dto

data class CitoyenDto(
    val cin: String? = null,
    val nomFr: String? = null,
    val nomAr: String? = null,
    val prenomFr: String? = null,
    val prenomAr: String? = null,
    val adresse: String? = null,
    val email: String? = null,
    val telephone: String? = null,
    val dateNaissance: String? = null,
    val lieuNaissance: String? = null,
    val sexe: String? = null,
    val photoUrl: String? = null
)
